tests/foreman/cli/test_variables.py

```diff
@@ -0,0 +1,1120 @@
+
+        @steps:
+
+        1.  Create a matcher with value that matches the list of step 2.
+        2.  Validate this value with list validator type and rule.
+
+        @assert: Matcher is created for matched validator rule.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_validate_matcher_value_with_default_type(self):
+        """Matcher is not created for value not of default type.
+
+        @id: 466197ea-44f0-46d0-b111-686b72183fe5
+
+        @steps:
+
+        1.  Create variable with valid default value.
+        2.  Create matcher with value that doesn't match the default type.
+
+        @assert: Matcher is not created for unmatched type.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_validate_matcher_value_with_default_type(self):
+        """Matcher is created for default type value.
+
+        @id: 033bf7d8-a488-49c1-b900-9e7169e945e0
+
+        @steps:
+
+        1.  Create variable default type with valid value.
+        2.  Create a matcher with value that matches the default type.
+
+        @assert: Matcher is created for matched type.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_validate_matcher_non_existing_attribute(self):
+        """Test matcher creation for non-existing attribute.
+
+        @id: 27ef1ef0-1c89-47eb-89e0-3da161154513
+
+        @steps:
+
+        1.  Attempt to create a matcher with non existing attribute.
+
+        @assert: Matcher is not created for non-existing attribute.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @skip_if_bug_open('bugzilla', 1259174)
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher(self):
+        """Create a Smart Variable with matcher.
+
+        @id: 42113584-d2db-4b91-8775-06bffee36be4
+
+        @steps:
+
+        1. Create a smart variable with valid name and default value.
+        2. Create a matcher for Host with valid value.
+
+        @assert:
+
+        1. The smart Variable with matcher is created successfully.
+        2. The variable is associated with host with match.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_attribute_priority(self):
+        """Matcher Value set on Attribute Priority for Host.
+
+        @id: 65144295-f0ca-4bd0-ae01-96c50ca829fe
+
+        @steps:
+
+        1.  Create variable with some default value.
+        2.  Set fqdn as top priority attribute.
+        3.  Create first matcher for fqdn with valid details.
+        4.  Create second matcher for some attribute with valid details.
+        Note - The fqdn/host should have this attribute.
+        5.  Go to YAML output of associated host.
```

Comment:
You are correct ! But I was thinking to newly introduce UI in CLI. There may be some features in future where we do something in CLI and we may need to check that in UI.
- OR -
 may be we need to find out where that YAML output generates locally.